### DISCUSSION TITLE
Laravel Nova 4 do not redirecting to login page if session expires or if clearing browsing history

### DISCUSSION THREAD

**Discussion by Pamposgsk** [laravel/nova-issues#4338] [Q&A] [answered]
- Laravel Version:9.0
- Nova Version: 4.7.0
- PHP Version: 8.0

Hi,
In laravel Nova 4 when the session expires or when clearing history the application do not redirecting to login page which is a real problem.
Kindly confirm the bug and let me know if this can be fixed
Thanks

**Comment by crynobone** [comment]
We already have dusk tests covering the use case: https://github.com/laravel/nova-dusk-suite/blob/cdeec6d57f22b308cbd07568d6853598be404f4e/tests/Browser/AuthenticatesUserTest.php#L69-L107

Will need full detail on how to reproduce the issue before we can consider this as a bug.

**Reply by crynobone** [reply]
Also, use the correct Bug Report template when submitting a bug report: https://github.com/laravel/nova-issues/issues/new?assignees=&labels=&template=1_Bug_report.md

**Comment by Pamposgsk** [comment]
Actually you can reproduce that by following the steps below:
1)After login Navigate to a custom tool of a laravel nova 4 project.
2)Clear history and website data
3)Refresh the page

you will get 404 error
In that case laravel nova 3 was redirecting to login page

**Comment by crynobone** [comment]
https://user-images.githubusercontent.com/172966/170984188-479134e1-d27c-46c4-b3cd-1b27324f9633.mp4

Unable to reproduce, the page will automatically redirect to login on the next notification polling interval or clicking on any page.

**Comment by Pamposgsk** [comment, marked as answer]
Let me tell you another way to reproduce it:

Lets say you have a resource named customers and a nova tool named Charts

1)Open google chrome and add the following url in the address bar : 127.0.0.1:8000/nova/resources/customers
It will automatically redirect you to : 127.0.0.1:8000/nova/login 
2) Open google chrome and add the following url in the address bar : 127.0.0.1:8000/nova/charts
You will get 404 error "page not found"

So in step 1 the redirection works as expected but when you try to navigate to a tool address the redirection not working

---------

# Solution

![image](https://user-images.githubusercontent.com/172966/171122073-a97dd406-5b23-4c38-b850-42004fd10faf.png)

You can add `Laravel\Nova\Http\Middleware\Authenticate` middleware to Tool's inertia routes to redirect guest user to login page instead of showing 404 error page.

**Reply by crynobone** [reply]
![image](https://user-images.githubusercontent.com/172966/171122073-a97dd406-5b23-4c38-b850-42004fd10faf.png)

You can add `Laravel\Nova\Http\Middleware\Authenticate` middleware to Tool's inertia routes to redirect guest user to login page instead of showing 404 error page.

**Reply by Pamposgsk** [reply]
Fixed!!!
Thanks a lot
